symforce/slam/imu_preintegration/imu_factor.cc

```diff
@@ -0,0 +1,52 @@
+/* ----------------------------------------------------------------------------
+ * SymForce - Copyright 2022, Skydio, Inc.
+ * This source code is under the Apache 2.0 license found in the LICENSE file.
+ * ---------------------------------------------------------------------------- */
+
+#include "./imu_factor.h"
+
+#include <Eigen/Cholesky>
+
+#include <sym/factors/internal_imu_factor.h>
+
+#include "preintegrated_imu_measurements.h"
+
+namespace sym {
+
+template <typename Scalar>
+ImuFactor<Scalar>::ImuFactor(const ImuPreintegrator<Scalar>& preintegrator)
+    : preintegrated_measurements_{preintegrator.PreintegratedMeasurements()},
+      // NOTE(brad, chao): llt then inverse is 2x faster than inverse then llt
+      sqrt_info_{preintegrator.Covariance().llt().matrixL().solve(
+          Eigen::Matrix<Scalar, 9, 9>::Identity())} {}
+
+template <typename Scalar>
+sym::Factor<Scalar> ImuFactor<Scalar>::Factor(const std::vector<Key>& keys_to_func) {
+  const auto begin = keys_to_func.begin();
+  return sym::Factor<Scalar>::Hessian(*this, keys_to_func, std::vector<Key>(begin, begin + 6));
```

Comment:
It might be good to note here that this copies `*this` into the returned factor - also I think this method should be `const`?  The magic number 6 maybe isn't ideal but I think it's probably clear enough

Reply:
Adding a note to me seems fine. And yup, should be const. And I did worry about the magic number a bit. How 'bout I mention it in the note?
```
template <typename Scalar>
sym::Factor<Scalar> ImuFactor<Scalar>::Factor(const std::vector<Key>& keys_to_func) const {
  const auto begin = keys_to_func.begin();
  // NOTE(brad): *this is copied. Keys to optimize happen to be first 6 keys to func
  return sym::Factor<Scalar>::Hessian(*this, keys_to_func, std::vector<Key>(begin, begin + 6));
}
```

Reply:
:+1: LGTM